File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> num_occurrences = new HashMap();
        for(int num: arr){
            num_occurrences.put(num, num_occurrences.getOrDefault(num, 0)+1);
        }
        HashSet<Integer> unique_vals = new HashSet(num_occurrences.values());
        return num_occurrences.size() == unique_vals.size();
    }
}

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        count = {}
        for i in range(len(arr)):
            if arr[i] in count:
                count[arr[i]] = count.get(arr[i],0)+1
            else:
                count[arr[i]] = 1
        seen = set()
        for values in count.values():
            if values in seen:
                return False
            else:
                seen.add(values)
        return True
